akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/EffectfulActorContext.scala

```diff
@@ -47,9 +45,9 @@ import scala.compat.java8.FunctionConverters._
     effectQueue.offer(MessageAdapter(implicitly[ClassTag[U]].runtimeClass.asInstanceOf[Class[U]], f))
     ref
   }
-  override def messageAdapter[U](messageClass: Class[U], f: function.Function[U, T]): ActorRef[U] = {
+  override def messageAdapter[U](messageClass: Class[U], f: akka.japi.function.Function[U, T]): ActorRef[U] = {
```

Comment:
let's discuss this, and change in separate PR. I see that we already use japi in `ReceiveBuilder` but I'm not sure we use it consistently everywhere and also if we would prefer to use standard Function despite inconvenience of checked exceptions (it's a standard library decision after all).

Reply:
I'll move it to a separate PR for discussion

Reply:
#27790